pkg/v3/stores/proposal_queue.go

```diff
@@ -47,8 +47,11 @@ func (pq *proposalQueue) Enqueue(newProposals ...ocr2keepers.CoordinatedBlockPro
 	defer pq.lock.Unlock()
 
 	for _, p := range newProposals {
-		if _, ok := pq.records[p.WorkID]; ok {
-			continue
+		if existing, ok := pq.records[p.WorkID]; ok {
+			if existing.proposal.Trigger.BlockNumber >= p.Trigger.BlockNumber {
```

Comment:
Can any of these fields be nil?

Reply:
good question! none of them are pointers so we should be good